org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/DOMCompletionEngine.java

```diff
@@ -682,4 +701,20 @@ private CompletionProposal toModuleCompletion(String moduleName, char[] prefix)
 		proposal.setRequiredProposals(new CompletionProposal[0]);
 		return proposal;
 	}
+
+	public ITypeBinding resolveBindings(IType type) {
+		ASTParser parser = ASTParser.newParser(AST.getJLSLatest());
+		parser.setSource(type.getTypeRoot());
+		parser.setBindingsRecovery(true);
+		parser.setResolveBindings(true);
+		parser.setStatementsRecovery(false);
+		CompilationUnit cu = (CompilationUnit) parser.createAST(null);
+		TypeDeclaration typeDecl = (TypeDeclaration) cu.findDeclaringNode(type.getKey());	
+		return typeDecl.resolveBinding();		
+	}
```

Comment:
This would be a very expensive operation to reparse every compilation unit. Moreover, this wouldn't work with binary types. I don't think it's suitable here.